### PR TITLE
Update packages and code

### PR DESCRIPTION
**Pull request by csc-felipe**
Updated the code to work with our setup.

- Dependencies updated
- Compile the plugin with Go from official Docker Alpine image
- Pin the Vault image to a specific version
- Ran `golangci-lint` and fixed raised issues
- Added `Step.GetData` as an alternative way of generating data in a Step

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# Debugging
+
+Set `SkipTeardown` to `true`, and go into the container with `docker exec -it container-id sh`. Find the container ID with `docker ps`.
+
+When leaving the container running, it's also possible to get the root token, and make API calls directly to Vault.
+For that, run execute `stepwise.Run` instead, create an environment, run it's `Setup()` function, and print the `env.RootToken()` to the command line.
+
+# Licensing
+
+`vault-testing-stepwise` is licensed under MRL-2.0.
```
Comment (blankdots) [inline README.md:75]:
```suggestion
`vault-testing-stepwise` is licensed under MPL-2.0.
```

File: environments/docker/environment.go [hunk 2]
```diff
@@ -463,10 +465,10 @@ func NewEnvironment(name string, options *stepwise.MountOptions) *DockerCluster
 		panic(err)
 	}
 
-	return &DockerCluster{
+	return &Cluster{
 		PluginName:   options.PluginName,
 		ClusterName:  fmt.Sprintf("test-%s-%s", name, clusterUUID),
-		RaftStorage:  true,
+		RaftStorage:  false,
```
Comment (blankdots) [inline environments/docker/environment.go:471]: is there an issue if we leave it to true ?

**Review by blankdots — COMMENTED** [review]
small typo.

overall very good work :rocket:

File: environments/docker/environment.go (hunk 2 above)
Comment (csc-felipe) [reply environments/docker/environment.go:471]: When debugging, I thought perhaps the raft storage was causing issues, but it actually works fine. I'll change it to true

**Comment by csc-felipe** [comment]
It was missing to update the package name as well 😆

**Review by blankdots — APPROVED** [review]

**Pull request merged**